pyprocessing/primitives_2d.py

```diff
@@ -1,6 +1,57 @@
 from pyprocessing import pp
 
 
+def ellipse(x, y, width, height):
+    global pp
+    pp.windows.set_ellipse(x, y, width, height)
+
+
+def rect(x, y, width, height, *args):
+    global pp
+    if len(args) == 0:
+        pp.windows.set_rectangle(x, y, width, height)
+    elif len(args) == 1:
+        pp.windows.set_rounded_rectangle(x, y, width, height,
+                                         args[0], args[0], args[0], args[0])
+    elif len(args) == 4:
+        pp.windows.set_rounded_rectangle(x, y, width, height,
+                                         args[0], args[1], args[2], args[3])
+    else:
+        raise TypeError
```

Comment:
When raising an Exception, always add a message for the developer to find what's wrong.

The else clause is unnecessary

I'd raise a ValueError over a TypeError here, but that can be discussed

Reply:
What do you mean the else is unnecessary? Without it the error would always be raised, which you don't want, and there is no behaviour for different numbers of arguments, so an error would have to be raised.

Reply:
My bad, you're right, I thought you returned after every condition.

Reply:
The other comments still hold true though